server/game/core/ability/abilityTargets/TargetResolverBaseClass.ts

```diff
@@ -0,0 +1,55 @@
+import { ITargetResolverBase } from '../../../TargetInterfaces';
+import { AbilityContext } from '../AbilityContext';
+import * as Contract from '../../utils/Contract';
+import { GameSystem } from '../../gameSystem/GameSystem';
+import PlayerOrCardAbility from '../PlayerOrCardAbility';
+import { RelativePlayer } from '../../Constants';
+
+/**
+ * Base class for all target resolvers.
+ */
+export abstract class TargetResolverBaseClass<TProps extends ITargetResolverBase<AbilityContext>> {
```

Comment:
Typically we don't include "Class" in the name of a class, but in SWU it's always weird because "Base" is a type of card so it's ambiguous. Maybe the cleanest way is just to put "TargetResolver" and let the `abstract` make it clear that it's a base class.